README.md

```diff
@@ -2,35 +2,65 @@
 
-PRs and feedback are welcomed. A good place to start is to review the current [open issues](https://github.com/DemocracyEarth/contracts/issues), please feel free to engage with us by commenting in a specific issue. Remember you can also check out the [Waffle board](https://waffle.io/DemocracyEarth/contracts) to understand what's being worked on now and what is coming up next.
+## Contributing
 
-## How to test locally
+PRs and feedback are welcomed. A good place to start is to review the current [open issues](https://github.com/DemocracyEarth/dapp/issues), please feel free to engage with us by commenting in a specific issue. Remember you can also check out the [Waffle board](https://waffle.io/DemocracyEarth/contracts) to understand what's being worked on now and what is coming up next.
 
-Project is being developed with NodeJs and Truffle, please make sure you have the latest version of npm before continuing. First follow the instructions to install [Truffle](http://truffleframework.com/):
+Dapp development is discussed actively in the slack channel [#developers](https://democracyearth.slack.com/messages/C5KCH0PD1) in democracyearth.slack.com 
 
-```sh
-$ npm install -g truffle
-```
+The dapp consists of mailnly two elements, contracts and frontend, located respectively in the following folders:
+- **contracts**: Solidity contracts, to be deployed to a local or remote blockchain.
+- **dApp-Srv-Exp**: JS frontend built using npm, bower and http-server
 
-Then clone or fork the repo, do an `npm install`, and you can start running the tests. You will need to run `truffle develop` to start a testing blockchain and then `test` to execute testing files. Please read the [Truffle docs](http://truffleframework.com/docs/getting_started/testing) to learn more about the tool.
+## Development dependencies
 
-## Compile contracts
+- NodeJS (make sure you have the latest version of npm, using `npm i -g npm`)
+  - Bower (`npm install -g bower`)
+  - Truffle (`npm install -g truffle`)
+  - http-server (`npm install -g http-server`)
+- Ganache ([http://truffleframework.com/ganache/]())
+- Metamask browser extension ([https://metamask.io/]())
 
-```sh
-truffle compile
-```
+## How to run
 
-## How to deploy the contracts
+### Compile, deploy and copy compiled contracts to frontend
+
+Choose the environment.
+Use *development* for local environment (start ganache before) or *ropsten* for Ropsten Ethereum test network in remote environment.
+
+ ```sh
+ $ npm install
+ $ truffle compile
+ $ truffle migrate --network [development / ropsten]
+ $ cp build/contracts/* dApp-Svr-Exp/contracts/
+ ```
+ 
+### Start frontend
 
 ```sh
-$ truffle migrate --network ropsten
-cp build/contracts/* dApp-Svr-Exp/contracts/
+$ http-server dApp-Svr-Exp/
```

Comment:
missed step `bower install` inside `dApp-Srv-Exp`, we can do it from main folder, but we need to add .bowerrc file

Reply:
for me it works just by doing npm install outside the dApp-Srv-Exp folder, shouldn't that be enough?

Reply:
I removed `bower_components` folder, then did an `npm_install`, and that recreated `bower_components` folder. So should we just leave `npm_install` and not add `bower install` for simplicity?